src/pruna/evaluation/metrics/metric_memory.py

```diff
@@ -100,23 +106,37 @@ class GPUMemoryMetric(BaseMetric):
         List of GPU indices to monitor. If None, all GPUs are assumed.
     """
 
-    def __init__(self, mode: str = "disk", gpu_indices: Optional[List[int]] = None) -> None:
+    def __init__(self, mode: str = DISK_MEMORY, gpu_indices: Optional[List[int]] = None) -> None:
         """
-        Initialize the GPUMemoryMetric.
+        Initialize the GPUMemoryStats.
 
         Raises
         ------
         ValueError
             If the provided mode is invalid.
         """
-        super().__init__()
         if mode not in VALID_MODES:
-            raise ValueError(f"Mode must be one of {VALID_MODES}, got '{mode}'.")
+            if mode in DEPRECATED_MODES:
+                warn(
+                    "GPUMemoryStats is no longer the preferred interface for GPU memory evaluation. "
+                    "Please use 'DiskMemoryMetric', 'InferenceMemoryMetric' or 'TrainingMemoryMetric' instead.",
+                    DeprecationWarning,
+                    stacklevel=2,
+                )
+                warn(
+                    f"Mode '{mode}' is deprecated and will be removed in a future release. "
+                    f"Please use '{DEPRECATED_MODES_MAP[mode]}' instead.",
+                    DeprecationWarning,
+                    stacklevel=2,
+                )
```

Comment:
arent we using `logger.warn`?

Reply:
I agree. I just thought I saw logger.warn somewhere else :) So, let's use `warn.warn`:)